cdap-ui/app/cdap/components/HeaderActions/index.js

```diff
@@ -17,23 +17,59 @@
 import React, {Component} from 'react';
 import { Dropdown, DropdownMenu, DropdownItem } from 'reactstrap';
 import cookie from 'react-cookie';
+import {Link} from 'react-router';
 import PlusButton from '../PlusButton';
 import T from 'i18n-react';
-
 require('./HeaderActions.less');
 var classNames = require('classnames');
-
+import Store from '../../services/store/store.js';
+const shortid = require('shortid');
 
 export default class HeaderActions extends Component {
 
   constructor(props) {
     super(props);
     this.state = {
       settingsOpen : false,
-      name : cookie.load('CDAP_Auth_User')
+      name : cookie.load('CDAP_Auth_User'),
+      namespaceList : [],
+      namespaceOpen : false,
+      currentNamespace: Store.getState().selectedNamespace
     };
     this.logout = this.logout.bind(this);
     this.toggleSettingsDropdown = this.toggleSettingsDropdown.bind(this);
+    this.namespaceList = [];
+    this.namespaceMap = '';
+    this.toggleNamespaceDropdown = this.toggleNamespaceDropdown.bind(this);
+    this.loadNamespacesFromStore = this.loadNamespacesFromStore.bind(this);
+    this.selectNamespaceAndCloseDropdown = this.selectNamespaceAndCloseDropdown.bind(this);
+  }
+
+  componentWillMount(){
+    //Load the namespaces into the dropdown from the store
+    this.namespaceList = Store.getState().namespaces;
+    if(this.namespaceList){
+      this.loadNamespacesFromStore(this.namespaceList, () => {
+        return Store.getState().selectedNamespace;
+      });
+    }
+
+    //Load updated data into the namespace dropdown
+    Store.subscribe(() => {
+
+      this.namespaceList = Store.getState().namespaces;
+      var namespaceToUpdate = Store.getState().selectedNamespace;
+
+      if(this.namespaceList){
+        this.loadNamespacesFromStore(this.namespaceList, () => {
```

Comment:
Why do we need to have the second argument as an empty callback function? Any reason why we can't have `namespaceToUpdate` as the second argument here?